ilastik/shell/gui/ilastikShell.py

```diff
@@ -1755,6 +1745,17 @@ def ensureNoCurrentProject(self, assertClean=False):
 
         return closeProject
 
+    def _save_current_applet(self):
+        projectFile = self.projectManager.currentProjectFile
+        if (
+            not self.projectManager.closed
+            and projectFile is not None
+            and not self.projectManager.currentProjectIsReadOnly
+        ):
+            if "currentApplet" in list(projectFile.keys()):
+                del projectFile["currentApplet"]
+            self.projectManager.currentProjectFile.create_dataset("currentApplet", data=self.currentAppletIndex)
```

Comment:
```suggestion
        pm = self.projectManager
        if pm.closed or pm.currentProjectIsReadOnly or pm.currentProjectFile is None:
            return
        ds = pm.currentProjectFile.require_dataset("currentApplet", shape=(), dtype=int)
        ds[()] = self.currentAppletIndex
```

Reply:
What does an empty tuple as an index do? I find this notation rather obscure and would prefer to avoid it.

Reply:
[This is now you use scalar datasets in HDF5](https://docs.h5py.org/en/stable/high/dataset.html#reading-writing-data). It looks weird, but actually makes sense: scalar dataset is 0-dimensional, so you use 0-dim tuple to reference a value.

Reply:
Heya, I agree wtith both of you :). Probably the right thing to use would be `require_dataset`, despite it's weird syntax... In ilastik `require_dataset` has not been used before - the other pattern is all over the place. So for now I'll keep the ilastik way of "requireing a dataset". We can discuss a refactor of the codebase to the different pattern of course.

The change to the conditions make a lot of sense, I'll  add those.

Reply:
One could also `group.pop("name", None)` instead of `if "name" in group: del group["name"]`, but we use the latter throughout ilastik too.